omni-reader/README.md

```diff
@@ -0,0 +1,224 @@
+  # Or specify multiple models as a list (new approach)
+  models: ["llama3.2-vision:11b", "mistral/pixtral-12b-2409"]
+  ground_truth_model: "gpt-4o-mini" # Model to use for ground truth when source is "openai"
+
+# Ground truth configuration
+ground_truth:
+  source: "openai" # Source: "openai", "manual", "file", or "none"
+  texts: [] # Ground truth texts (for manual source)
+  file: null # Path to ground truth JSON file (for file source)
+
+# Output configuration
+output:
+  ground_truth:
+    save: false # Whether to save ground truth data
+    directory: "ocr_results" # Directory for ground truth data
+
+  ocr_results:
+    save: false # Whether to save OCR results
+    directory: "ocr_results" # Directory for OCR results
+
+  visualization:
+    save: false # Whether to save HTML visualization
+    directory: "visualizations" # Directory for visualization
+```
+
+### Running with Command Line Arguments
+
+You can still use command line arguments for quick runs:
+
+```bash
+# Run with default settings (processes all images in assets directory)
+python run.py --image-folder assets
+
+# Run with custom prompt
+python run.py --image-folder assets --custom-prompt "Extract all text from this image."
+
+# Run with specific images
+python run.py --image-paths assets/image1.jpg assets/image2.png
+
+# Run with OpenAI ground truth for evaluation
+python run.py --image-folder assets --ground-truth openai --save-ground-truth
+
+# List available ground truth files
+python run.py --list-ground-truth-files
+
+# For quicker processing of a single image without metadata or artifact tracking
+python run_compare_ocr.py --image assets/your_image.jpg --model both
+
+# Run comparison with multiple specific models in parallel
+python run_compare_ocr.py --image assets/your_image.jpg --model "gemma3:12b,llama3.2-vision:11b,moondream"
+
+# Run comparison with all available models in parallel
+python run_compare_ocr.py --image assets/your_image.jpg --model all
+```
+
+### Using the Streamlit App
+
+For interactive use, the project includes a Streamlit app:
+
+```bash
+streamlit run app.py
+```
+
+<div align="center">
+  <img src="assets/streamlit.png" alt="Streamlit UI Interface" width="800"/>
+  <p><em>Interactive Streamlit interface for easy document processing and model comparison</em></p>
+</div>
+
+### ### Remote Artifact Storage and Execution
+
+This project supports storing artifacts remotely and executing pipelines on cloud infrastructure. Follow these steps to configure your environment for remote operation:
+
+#### Setting Up Cloud Provider Integrations
+
+Install the appropriate ZenML integrations for your preferred cloud provider:
+
+```bash
+# For AWS
+zenml integration install aws s3 -y
+
+# For Azure
+zenml integration install azure azure-blob -y
+
+# For Google Cloud
+zenml integration install gcp gcs -y
+```
+
+For detailed configuration options and other components, refer to the ZenML documentation:

```

Comment:
id link https://docs.zenml.io/how-to/popular-integrations/aws-guide, https://docs.zenml.io/how-to/popular-integrations/gcp-guide, and https://docs.zenml.io/how-to/popular-integrations/azure-guide